idaes/apps/flexibility_analysis/decision_rules/linear_dr.py

```diff
@@ -0,0 +1,127 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+This module contains a function for constructing a linear decision rule for 
+the inner problem of the flexibility test.
+"""
+from typing import MutableMapping, Sequence
+import pyomo.environ as pe
+from pyomo.core.base.var import _GeneralVarData, ScalarVar, IndexedVar
+from pyomo.core.base.block import _BlockData
+from pyomo.core.expr.numeric_expr import LinearExpression
+from pyomo.common.config import ConfigValue
+from .dr_config import DRConfig
+from ..check_optimal import assert_optimal_termination
+
+
+class LinearDRConfig(DRConfig):
+    r"""
+    A class for specifying options for constructing linear decision rules
+    for use in the flexibility test problem.
+
+    Attributes
+    ----------
+    solver: Union[Solver, OptSolver]
+        The solver to use for building the linear decision rule (an LP solver).
+    """
+
+    def __init__(
+        self,
+        description=None,
+        doc=None,
+        implicit=False,
+        implicit_domain=None,
+        visibility=0,
+    ):
+        super().__init__(
+            description=description,
+            doc=doc,
+            implicit=implicit,
+            implicit_domain=implicit_domain,
+            visibility=visibility,
+        )
+        self.solver = self.declare(
+            "solver", ConfigValue(default=pe.SolverFactory("ipopt"))
```

Comment:
Should this also have a `doc` attribute, just in case someone tries to query the ConfigDict?

Reply:
Absolutely. I'll add one.